src/bio/bio_xstream.c

```diff
@@ -195,9 +195,9 @@ set_faulty_criteria(void)
 	 */
 	glb_criteria.fc_max_csum_errs = UINT32_MAX;
 
-	d_getenv_bool("DAOS_NVME_AUTO_FAULTY_ENABLED", &glb_criteria.fc_enabled);
-	d_getenv_int("DAOS_NVME_AUTO_FAULTY_IO", &glb_criteria.fc_max_io_errs);
-	d_getenv_int("DAOS_NVME_AUTO_FAULTY_CSUM", &glb_criteria.fc_max_csum_errs);
+	d_getenv_bool(&glb_criteria.fc_enabled, "DAOS_NVME_AUTO_FAULTY_ENABLED");
```

Comment:
- [ ] Revert interfaces update

Reply:
By any chance, do you know if we have some functional tests using MPI_IO ?

Reply:
You can use `Features: mpiio`

Reply:
In the past where this has come up then what functional testing we have uses a weird system where:

1. PRs and landings to downstream components test against master daos.
2. PRs and landings to daos test with the latest built downstream component.

The unfortunate implication here is that we have some testing but it's limited and if we change the API or ABI and it's incompatible then we don't notice until there's a new PR to the downstream component or it gets rebuilt for another reason.  If we had test coverage here then it would fail at build time, as it is it'll fail (obscurely) at runtime, but only if the code is called.

It's an area we need to work on.

Reply:
Fixed with commit ac764d169812b82850bd1bf1d9d9177415c2f857

Reply:
> * [x]  Revert interfaces update

Fixed with commit ac764d169812b82850bd1bf1d9d9177415c2f857